layersvt/api_dump.h

```diff
@@ -833,389 +851,359 @@ class ApiDumpInstance {
 
     bool conditional_initialized = false;
     bool should_dump_output = true;
-    bool first_func_call_on_frame = false;
+    bool first_func_call_on_frame = true;
 
     std::chrono::system_clock::time_point program_start;
 
     // Store the VkInstance handle so we don't use null in the call to
     // vkGetInstanceProcAddr(instance_handle, "vkCreateDevice");
     std::unordered_map<VkPhysicalDevice, VkInstance> vk_instance_map;
+
+    // Storage for getCmdBufferLevel() which is called in a place where it needs access to the cmd_buffer but it isn't present in
+    // the current structure.
+    VkCommandBuffer cmd_buffer;
+
+    // Storage for VkPipelineViewportStateCreateInfo which needs to ignore the scissor and viewport pipeline state if their
+    // respective dynamic state is set.
+    bool is_dynamic_scissor;
+    bool is_dynamic_viewport;
 };
 
 // Utility to output an address.
 // If the quotes arg is true, the address is encloded in quotes.
 // Used for text, html, and json output.
-void OutputAddress(const ApiDumpSettings &settings, const void *addr, bool quotes) {
-    if (quotes) settings.stream() << "\"";
+void OutputAddress(const ApiDumpSettings &settings, const void *addr) {
```

Comment:
Really not your code... but (almost?) everywhere I look, callers of `OutputAddress` first do a special case to check the address against `NULL`, and then typically short-circuit.

It's confusing to have both callers checking the address against `NULL` and also checking it here.  Some refactoring of this would be welcome.

Reply:
Looking at all the places OutputAddress is called, 7 places are guaranteed to be non-null, 4 are null, and 5 are unknowable at the call site. I see what you mean that they appear to be unnecessary null checks, but they are indeed necessary (in some places that is).